R/tree.R

```diff
@@ -31,200 +31,284 @@
+        ## Comparison of different nucleotide or amino acid substitution models
+        mt <- modelTest(prot_subset, model = model_test)
+        print(mt)
 
-    # estimate a distance matrix using a Jules-Cantor Model
-    dna_dist <- dist.ml(prot10, model = "JC69")
+        # estimate a distance matrix using a Jules-Cantor Model
+        dna_dist <- dist.ml(prot_subset, model = dna_model)
 
-    ## Neighbor Joining, UPGMA, and Maximum Parsimony
-    # quick and dirty UPGMA and NJ trees
-    prot_UPGMA <- upgma(dna_dist)
-    prot_NJ <- NJ(dna_dist)
-    plot(prot_UPGMA, main = "UPGMA")
-    plot(prot_NJ, main = "Neighbor Joining")
+        ## Neighbor Joining, UPGMA, and Maximum Parsimony
+        # quick and dirty UPGMA and NJ trees
+        prot_UPGMA <- upgma(dna_dist)
+        prot_NJ <- NJ(dna_dist)
+        plot(prot_UPGMA, main = "UPGMA")
+        plot(prot_NJ, main = "Neighbor Joining")
 
-    # parsimony searches
-    prot_optim <- optim.parsimony(prot_NJ, prot)
-    prot_pratchet <- pratchet(prot10) # returning error
-    plot(prot_optim)
-    plot(prot_pratchet)
+        # parsimony searches
+        prot_optim <- optim.parsimony(prot_NJ, prot_phyDat)
+        prot_pratchet <- pratchet(prot_subset) # returning error
+        plot(prot_optim)
+        plot(prot_pratchet)
 
-    ## Maximum likelihood and Bootstrapping
-    # ml estimation w/ distance matrix
-    fit <- pml(prot_NJ, prot10)
-    print(fit)
-    fitJC <- optim.pml(fit, model = "JC", rearrangement = "stochastic")
-    logLik(fitJC)
-    bs <- bootstrap.pml(fitJC,
-        bs = 100, optNni = TRUE, multicore = TRUE,
-        control = pml.control(trace = 0)
-    )
-    plotBS(midpoint(fitJC$tree), bs, p = 50, type = "p")
+        ## Maximum likelihood and Bootstrapping
+        # ml estimation w/ distance matrix
+        fit <- pml(prot_NJ, prot_subset)
+        print(fit)
+        fitJC <- optim.pml(fit, model = ml_model, rearrangement = rearrangement)
+        logLik(fitJC)
+        bs <- bootstrap.pml(fitJC,
+            bs = bootstrap_reps, optNni = TRUE, multicore = TRUE,
+            control = pml.control(trace = 0)
+        )
+        plotBS(midpoint(fitJC$tree), bs, p = bootstrap_p, type = plot_type)
 
-    # subsetted alignment bs example
-    prot10_dm <- dist.ml(prot10)
-    prot10_NJ <- NJ(prot10_dm)
-    fit2 <- pml(prot10_NJ, data = prot10)
-    print(fit2)
-    fitJC2 <- optim.pml(fit2, model = "JC", rearrangement = "stochastic")
-    logLik(fitJC2)
-    bs_subset <- bootstrap.pml(fitJC2,
-        bs = 100, optNni = TRUE, multicore = TRUE,
-        control = pml.control(trace = 0)
-    )
-    bs2 <- plotBS(midpoint(fitJC2$tree), bs, p = 50, type = "p")
+        # subsetted alignment bs example
+        prot_subset_dm <- dist.ml(prot_subset)
+        prot_subset_NJ <- NJ(prot_subset_dm)
+        fit2 <- pml(prot_subset_NJ, data = prot_subset)
+        print(fit2)
+        fitJC2 <- optim.pml(fit2, model = ml_model, rearrangement = rearrangement)
```

Comment:
```suggestion
        fit_optimized2 <- optim.pml(fit2, model = fit2, rearrangement = rearrangement)
```

As before, if `model` should be the output of `pml()` then update. Discard if I'm mistaken though!